fake_spectra/abstractsnapshot.py

```diff
@@ -325,12 +361,13 @@ def _segment_to_partlist(self, part_type, segment):
         """Get the first and last particle in a segment."""
         if segment < 0:
             return (0, None)
-        n_segments = self.get_n_segments()
-        one_segment = int(self.get_npart()[part_type]//n_segments)
-        if segment < n_segments -1:
-            return (one_segment*segment, one_segment*(segment+1))
-        #Last segment has no end
-        return (one_segment*segment, None)
+        n_segments = self.get_n_segments(part_type)
+        one_segment = int(self.parts_rank[self.rank]/n_segments)
+        if self.rank ==0 :
```

Comment:
Comment here would be useful too: it isn't immediately obvious although I figured it out.